app/v2/controllers/ErrorTranslator.scala

```diff
@@ -19,14 +19,26 @@ package v2.controllers
 import cats.data.EitherT
 import cats.data.NonEmptyList
 import cats.effect.IO
+import uk.gov.hmrc.http.HeaderCarrier
+import v2.models.AuditInfo
 import v2.models.errors._
+import v2.services.AuditService
 
 trait ErrorTranslator {
 
   implicit class ErrorConverter[E, A](value: EitherT[IO, E, A]) {
 
-    def asPresentation(implicit c: Converter[E]): EitherT[IO, PresentationError, A] =
-      value.leftMap(c.convert)
+    def asPresentation(auditInfo: AuditInfo, auditService: AuditService)(implicit
+      c: Converter[E],
+      headerCarrier: HeaderCarrier
+    ): EitherT[IO, PresentationError, A] = {
+      implicit val info = auditInfo
+      value.leftMap {
+        error =>
+          auditService.balanceRequestFailed(error)
```

Comment:
We're calling balanceRequestFailed here and in requestFailed - one seems to be for InvalidPayloadEvent and the other for various errors, can we rename them so it's less ambiguous?